### PR TITLE
chore: Move `overwrites.yaml` to config directory

### LINKED ISSUE #17164: update location of gossip overrides configuration.
The following changes need to be made to the process of configuring the gossip overrides. 
1. The location of the config file should move to `data/config` instead of the root directory. 
2. The name of the file needs to be updated to: `node-overrides.yaml` instead of `overwrites.yaml`
3. [The label in the config file](https://github.com/hashgraph/hedera-services/blob/7d83ba28c8ac0b0cecfeb2c1a306c85fd4c9568f/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/config/GossipConfig.java#L27) `networkEndpoints` should change to `interfaceBindings` (also rename the accessor method)

### PR DESCRIPTION
**Pull request by timo0**
Fix #17164 

* Renamed `overwrites.yaml` to `node-overrides.yaml` to clarify the purpose
* Moved from root directory to `data/config` directory 
* Renamed `networkEndpoints` to `interfaceBindings`

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/config/GossipConfig.java [hunk 1]
```diff
@@ -37,8 +37,8 @@ public record GossipConfig(
      * @param nodeId the node ID
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/config/GossipConfig.java:37]: The comments on this method need to be updated to reflect the field name change.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/config/GossipConfig.java [hunk 2]
```diff
@@ -24,7 +24,7 @@
 
 @ConfigData("gossip")
 public record GossipConfig(
-        @ConfigProperty(defaultValue = Configuration.EMPTY_LIST) List<NetworkEndpoint> networkEndpoints,
+        @ConfigProperty(defaultValue = Configuration.EMPTY_LIST) List<NetworkEndpoint> interfaceBindings,
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/config/GossipConfig.java:27]: Could be add  class level javadoc explaining the the meaning and purpose of each parameter?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformBuildConstants.java [hunk 3]
```diff
@@ -40,7 +40,7 @@ public final class PlatformBuildConstants {
     /**
      * The name of the default overwrites file (i.e. where interface or host overwrites can be specified).
      */
-    public static final String DEFAULT_OVERWRITE_YAML_FILE_NAME = "overwrites.yaml";
+    public static final String DEFAULT_OVERWRITE_YAML_FILE_NAME = "data/config/node-overwrites.yaml";
```
Comment (nathanklick) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformBuildConstants.java:43]:
```suggestion
    public static final String DEFAULT_OVERWRITE_YAML_FILE_NAME = "data/config/node-overrides.yaml";
```

**Review by nathanklick — CHANGES_REQUESTED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformBuildConstants.java (hunk 3 above)
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformBuildConstants.java]: added manually

**Review by edward-swirldslabs — APPROVED** [review]

**Review by nathanklick — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]
Services changes look good 👍

**Pull request merged**